src/main/java/com/synopsys/integration/alert/channel/jira/JiraIssueHandler.java

```diff
@@ -192,12 +169,64 @@ public JiraMessageResult createOrUpdateIssues(JiraIssueConfig jiraIssueConfig, M
+            if (jiraIssueConfig.getCommentOnIssues()) {
+                String operationComment = createOperationComment(operation, arbitraryItem.getCategory(), providerName, combinedItems);
+                addComment(issue.getKey(), operationComment);
+                updatedIssues.add(issue);
+            }
+
+            boolean didUpdateIssue = jiraTransitionHelper.transitionIssueIfNecessary(issue.getKey(), jiraIssueConfig, operation);
+            if (didUpdateIssue) {
+                updatedIssues.add(issue);
+            }
+        }
+        return updatedIssues;
+    }
+
+    private IssueResponseModel createIssue(IssueRequestModelFieldsBuilder initialFieldsBuilder, JiraIssueConfig jiraIssueConfig, String providerName, LinkableItem topic, Optional<LinkableItem> subTopic, ComponentItem arbitraryItem,
+        String trackingKey) throws IntegrationException {
+        initialFieldsBuilder.setProject(jiraIssueConfig.getProjectComponent().getId());
+        initialFieldsBuilder.setIssueType(jiraIssueConfig.getIssueType());
+        String issueCreator = jiraIssueConfig.getIssueCreator();
+        IssueResponseModel issue = null;
+        try {
+            issue = issueService.createIssue(new IssueCreationRequestModel(issueCreator, jiraIssueConfig.getIssueType(), jiraIssueConfig.getProjectComponent().getName(), initialFieldsBuilder, List.of()));
+            logger.debug("Created new Jira Cloud issue: {}", issue.getKey());
+        } catch (IntegrationRestException e) {
+            handleIssueCreationRestException(e, issueCreator);
+        }
+        if (issue == null || StringUtils.isBlank(issue.getKey())) {
+            throw new AlertException("There was an problem when creating this issue.");
+        }
+        String issueKey = issue.getKey();
+        addIssueProperties(issueKey, providerName, topic, subTopic, arbitraryItem, trackingKey);
+        addComment(issueKey, "This issue was automatically created by Alert.");
+        return issue;
+    }
+
+    private List<IssueComponent> retrieveExistingIssues(String jiraProjectKey, String provider, LinkableItem topic, Optional<LinkableItem> subTopic, ComponentItem componentItem, String alertIssueUniqueId) throws IntegrationException {
+        return jiraIssuePropertyHelper
+                   .findIssues(jiraProjectKey, provider, topic, subTopic.orElse(null), componentItem, alertIssueUniqueId)
                    .map(IssueSearchResponseModel::getIssues)
-                   .map(List::stream)
-                   .flatMap(Stream::findFirst);
+                   .orElse(List.of());
+    }
+
+    // Only the DELETE operation can update more than one issue.
+    // If an ADD or UPDATE operation has more than one issue it could operate on, then the issue properties are either not unique enough or the property indexer is not installed.
+    private List<IssueComponent> filterUpdatableIssues(List<IssueComponent> issues, ItemOperation operation) {
```

Comment:
Most of this method can be deleted and you can instead modify the list right before line 129 then pass the proper list through the methods instead of one that will just be modified later.

Reply:
I think the abstraction makes it more readable. This would have been > 4 levels of nesting.